dev/source/docs/ros2-sitl.rst

```diff
@@ -51,18 +128,22 @@ Once everything is running, you can now interact with ArduPilot through the ROS
     # Echo a topic published from ArduPilot
     ros2 topic echo /ap/geopose/filtered
 
-If the ROS 2 topics aren't being published, ensure the ardupilot parameter ref:`DDS_ENABLE<DDS_ENABLE>` is set to 1 and reboot the launch.
+DDS is responsible for ROS2 communication. Therefore, DDS must be enabled by setting the Ardupilot parameter `DDS_ENABLE` and make the parameter `DDS_DOMAIN_ID` match `ROS_DOMAIN_ID` of your ROS2 settings.
+
+For instance, if the ROS 2 topics aren't being published, first check if `DDS_ENABLE` is set to ``1`` or not. If not, set as ``1``. It can be done through Mission Planner, QGroundControl or the `param <https://ardupilot.org/mavproxy/docs/getting_started/cheatsheet.html>`__ command in the MAVProxy console. The second aspect to check, ensure the ArduPilot parameter `DDS_DOMAIN_ID` matches your environment variable ``ROS_DOMAIN_ID``. The default is ``0`` for ArduPilot, which corresponds to the environment variable being unset. You may need to relaunch the SITL after changing the parameters.
+
+To change the parameters using `mavproxy`, you can launch `mavproxy` in another terminal and connect to SITL through UDP
 
 .. code-block:: bash
 
     export PATH=$PATH:~/ardu_ws/src/ardupilot/Tools/autotest
-    sim_vehicle.py -w -v ArduPlane --console -DG --enable-dds
+    mavproxy.py --master udp:127.0.0.1:14550
 
     param set DDS_ENABLE 1
+    param set DDS_DOMAIN_ID 1 # if you set export ROS_DOMAIN_ID=1
```

Comment:
I would not put this here. People will blindly copy that. 

Previously, this code block was for `If the ROS 2 topics aren't being published, ensure the ardupilot parameter ref:`DDS_ENABLE<DDS_ENABLE>` is set to 1 and reboot the launch.`

Now, you are using it to show how to change parameter that most people don't need to do.

I think we should just delete the block now that DDS_ENABLE is defaulted to 1.